packages/protocol/src/utils/type-helper.ts

```diff
@@ -9,3 +9,9 @@ export type ValuesAsUnionDeep<T> = {
 declare const brand: unique symbol;
 
 export type Brand<T, TBrand> = T & { [brand]: TBrand };
+
+export type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void ? I : never;
```

Comment:
This converts a type of unions to an intersected type, e.g.
`{name: string} | {description: string}` becomes to 
`{name: string} & {description: string}` or better 
`{name: string; description: string}`